python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -239,3 +245,250 @@ def readFits(cls, filename):
+        return stats_control, stats_flags
+
+    def run(self, bss_ref_list, region_name=None):
+        """Read input bright star stamps and stack them together.
+
+        The stacking is done iteratively over smaller areas of the final model
+        image to allow for a great number of bright star stamps to be used.
+
+        Parameters
+        ----------
+        bss_ref_list : `list` of
+                `lsst.daf.butler._deferredDatasetHandle.DeferredDatasetHandle`
+            List of available bright star stamps data references.
+        region_name : `str`, optional
+            Name of the focal plane region, if applicable. Only used for
+            logging purposes, when running over multiple such regions
+            (typically from `MeasureExtendedPsfTask`)
+        """
+        log_message = f'Building extended PSF from stamps extracted from {len(bss_ref_list)} detector images '
+        if region_name:
+            log_message += f'for region "{region_name}".'
+        self.log.info(log_message)
+        # read in example set of full stamps
+        for bss_ref in bss_ref_list:
+            if not isinstance(bss_ref, DeferredDatasetHandle):
+                if not bss_ref.datasetExists("brightStarStamps"):
+                    self.log.warn("Could not find %s %s; skipping it", "brightStarStamps", bss_ref.dataId)
+                    bss_ref_list.remove(bss_ref)
+                    continue
+            bss = bss_ref.get(datasetType="brightStarStamps", immediate=True)
+            break
+        example_stamp = bss[0].stamp_im
+        # create model image
+        ext_psf = afwImage.MaskedImageF(example_stamp.getBBox())
+        # divide model image into smaller subregions
+        subregion_size = Extent2I(*self.config.subregion_size)
+        sub_bboxes = AssembleCoaddTask._subBBoxIter(ext_psf.getBBox(), subregion_size)
+        # compute approximate number of subregions
+        n_subregions = int(ext_psf.getDimensions()[0]/subregion_size[0] + 1)*int(
+            ext_psf.getDimensions()[1]/subregion_size[1] + 1)
+        self.log.info(f"Stacking will performed iteratively over approximately {n_subregions} "
+                      "smaller areas of the final model image.")
+        # set up stacking statistic
+        stats_control, stats_flags = self._set_up_stacking(example_stamp)
+        # perform stacking
+        for jbbox, bbox in enumerate(sub_bboxes):
+            all_stars = None
+            for bss_ref in bss_ref_list:
+                if not isinstance(bss_ref, DeferredDatasetHandle):
+                    if not bss_ref.datasetExists("brightStarStamps"):
+                        self.log.warn("Could not find %s %s; skipping it", "brightStarStamps", bss_ref.dataId)
+                        bss_ref_list.remove(bss_ref)
+                        continue
+                read_stars = bss_ref.get(datasetType="brightStarStamps", parameters={'bbox': bbox})
```

Comment:
As above, this might want to be wrapped in a try/except block. (However, if the `bss_ref_list` checking is moved to its own initial section, then you might be safe leaving this line as-is, due to the fact that you'd then be able to guarantee this dataset type definitely already exists in `bss_ref_list`.